src/integration-test/java/edu/kit/ipd/crowdcontrol/objectservice/mail/MailHandlerTest.java

```diff
@@ -75,5 +77,79 @@ public void test() throws Exception {
             }
         }
         assertTrue(found);
+
+        messages = fetcher.fetchFolder(folder);
+        found = false;
+        for (Message message : messages) {
+            if (message.getSubject().equals(subject)) {
+                found = true;
+                break;
+            }
+        }
+        assertFalse(found);
+    }
+
+    @Test
+    public void testFetchUnseenMarkAsUnseen() throws Exception {
+        String uuid = UUID.randomUUID().toString();
+        String subject = "[test] " + uuid;
+
+        sender.sendMail(receiver, subject, uuid);
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
+        }
+        Message[] messages = fetcher.fetchUnseen(folder);
+
+        boolean found = false;
+
+        Message testMessage = null;
+        for (Message message : messages) {
+            if (message.getSubject().equals(subject)) {
+                found = true;
+                testMessage = message;
+                break;
+            }
+        }
+        assertTrue(found);
```

Comment:
if you want, i can add a javadoc note, that if the tests won't work, you could set the sleeptime higher. (and then better refactor it to a constant)